### PR TITLE
adds array helper and tests

### LINKED ISSUE #142: (array ) helper maybe?
Sometimes it's helpful to compose an array in the template. 

What do you think about something like this?

```hbs
{{#each (array 'foo' 'bar' 'baz') as |word|}}
  {{word}}
{{/each}}
```

The implementation is like hash helper,

```javascript
export default Ember.Helper.helper(function(params) {
  return params;
});
```

If you agree, I'll make a PR. 

Also, if there is another way to do with that's already built-in, please let me know!

### PR DESCRIPTION
**Pull request by zigahertz**
closes #142 

Signed-off-by: Taras Mankovski <taras@embersherpa.com>

### CODE REVIEW

File: addon/helpers/array.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+export function array(params=[]) {
+  return params;
```
Comment (poteto) [inline addon/helpers/array.js]: If you wrap this with `Ember.A()`, does it still appear truthy in the template when empty?

File: addon/helpers/array.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import Ember from 'ember';
+
+const { A, isNone };
+
+export function array(params) {
+  if (isNone(params)) {
+    params = new A();
```
Comment (zigahertz) [inline addon/helpers/array.js]: @poteto it doesn't make a difference - but we added it here

File: addon/helpers/array.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Ember from 'ember';
+
+const { A, isNone };
```
Comment (poteto) [inline addon/helpers/array.js]: You can import it instead:

```js
import { A as emberArray } from 'ember-array/utils';
```

File: addon/helpers/array.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+export function array(params) {
+  if (isNone(params)) {
```
Comment (poteto) [inline addon/helpers/array.js]: This check isn't necessary, you can just return:

```js
return emberArray(params);
```

File: tests/integration/helpers/array-test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  this.set('dynamic', 'π');
+  assert.equal(this.$().text().trim(), '12π3', 'dynamic is included');
+});
+
+test('if evaluates empty array to false', function(assert) {
+  this.render(hbs`
+    {{~if (array) 'true' 'false'~}}
+  `);
+  assert.ok([], 'JavaScript evaluates empty array as true');
+  assert.equal(this.$().text().trim(), 'false', 'empty array evaluates to false');
```
Comment (poteto) [inline tests/integration/helpers/array-test.js]: I think this behavior is fine – it makes sense when it is in the context of a template. An empty array _should_ be falsey when used in hbs

File: tests/integration/helpers/array-test.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  this.set('dynamic', 'π');
+  assert.equal(this.$().text().trim(), '12π3', 'dynamic is included');
+});
+
+test('if evaluates empty array to false', function(assert) {
+  this.render(hbs`
+    {{~if (array) 'true' 'false'~}}
+  `);
+  assert.ok([], 'JavaScript evaluates empty array as true');
```
Comment (poteto) [inline tests/integration/helpers/array-test.js]: This is not a useful assertion, we can remove it

File: tests/integration/helpers/array-test.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  assert.equal(this.$().text().trim(), 'false', 'empty array evaluates to false');
+});
+
+test('if helper treats empty arrays as false', function(assert) {
```
Comment (poteto) [inline tests/integration/helpers/array-test.js]: I think this test is outside the scope of this helper and can be removed

**Comment by zigahertz** [comment]
@poteto thanks for your comments, just updated the code to reflect them.

**Comment by poteto** [comment]
Looks great @zigahertz! Could you please squash it down into a single commit? Thank you!

**Comment by zigahertz** [comment]
thanks @poteto, squash is done!

**Comment by poteto** [comment]
@homu r+
